File: src/com/leetcode/Easy/KidsWithTheGreatestNumberOfCandies.java
package com.leetcode.Easy;

import java.util.*;

// 1431. Kids With the Greatest Number of Candies
public class KidsWithTheGreatestNumberOfCandies {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        List<Boolean> result = new ArrayList<>();
        int max = Arrays.stream(candies).max().getAsInt();
        for (int candy : candies) {
            if (candy + extraCandies >= max)
                result.add(true);
            else
                result.add(false);
        }
        return result;
    }
}

File: JavaScript/Easy/KidsWithTheGreatestNumberOfCandies.js
/**
 * 1431. Kids With the Greatest Number of Candies
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function (candies, extraCandies) {
  const res = [];
  const max = Math.max(...candies);
  for (let i = 0; i < candies.length; i++) {
    if (candies[i] + extraCandies >= max) {
      res.push(true);
    } else {
      res.push(false);
    }
  }
  return res;
};
